relay-server/src/metrics_extraction/event.rs

```diff
@@ -47,31 +47,41 @@ impl Extractable for Span {
 /// If this is a transaction event with spans, metrics will also be extracted from the spans.
 pub fn extract_metrics(
     event: &Event,
+    spans_extracted: bool,
     config: &MetricExtractionConfig,
     max_tag_value_size: usize,
     span_extraction_sample_rate: Option<f32>,
 ) -> Vec<Bucket> {
     let mut metrics = generic::extract_metrics(event, config);
 
-    if !sample(span_extraction_sample_rate.unwrap_or(1.0)) {
-        return metrics;
+    // If spans were already extracted for an event,
+    // we rely on span processing to extract metrics.
+    if !spans_extracted && sample(span_extraction_sample_rate.unwrap_or(1.0)) {
```

Comment:
Unrelated thought: Should we change the signature of sample to: `fn sample(rate: impl Into<Option<f32>>) -> bool` instead?

Reply:
I kinda like the simple signature of `sample`, to be honest.